src/index.ts

```diff
@@ -312,37 +313,40 @@ function addBrowserCommands(
           return item;
         })
       );
-      const params = parseHdfQuery(items[0].path);
+      // Selected item takes precedence on current path
+      const selectedItem = items[0] || widget.model;
+      const params = parseHdfQuery(selectedItem.path);
 
       try {
         const attrs = await hdfAttrsRequest(params, serverSettings);
         const widget = new MainAreaWidget<AttributeViewer>({
-          content: new AttributeViewer(attrs)
+          content: new AttributeViewer(attrs),
         });
-        widget.title.label = items[0].name;
+        widget.title.label = nameFromPath(params.uri);
```

Comment:
When no item is selected, the only way to get the `name` is to deduce it from the `uri`